app/assets/javascripts/profile/authentication.coffee

```diff
@@ -77,6 +77,13 @@ class Identity
     # TODO: figure out a way for the BE to pass the url
     window.location.href = "/auth/#{@getType()}"
 
+  handleDelete: (response) ->
+    location = response['location']
+    if location
+      window.location.href = location
+    else
+      @moveToDisabledSection()
+
 class Password extends Identity
```

Comment:
The above could be arguably more `coffeescripty` by using dot notation and the existential operator `?`.
```coffee
handleDelete: (response) ->
    if response.location?
       window.location.href = response.location
    else
       @moveToDisabledSection()
```

There's nothing at all wrong with how it is though, it's still very clear.